### PR TITLE
test: Report test status success/failure back to Timestream

### PR DESCRIPTION
**Pull request by jonathan-innis**
Fixes #N/A 

**Description**

**How was this change tested?**

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

File: test/cloudformation/timestream_cloudformation.yaml [hunk 1]
```diff
@@ -52,6 +55,16 @@ Resources:
...
+  TestStatusTimestreamTable:
+    Type: "AWS::Timestream::Table"
+    Properties:
+      DatabaseName: !Ref "TimestreamDatabase"
+      TableName: !Sub "${TestStatusTableName}"
+      MagneticStoreWriteProperties:
+        EnableMagneticStoreWrites: true
+      RetentionProperties:
+        MemoryStoreRetentionPeriodInHours: "2160" # Three months of memory store retention
+        MagneticStoreRetentionPeriodInDays: "1825" # 5 years of magnetic store retention
```
Comment (rschalo) [inline test/cloudformation/timestream_cloudformation.yaml:67]: WDYT about 1 year?

Comment (jonathan-innis) [reply test/cloudformation/timestream_cloudformation.yaml:67]: We can probably drop it later. I don't think we've seen this come up as a big cost thing at this point (we don't have _that_ much data there)

**Pull request closed without merging**